zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/http/ElasticsearchHttpSpanConsumer.java

```diff
@@ -37,6 +38,15 @@
 
   final ElasticsearchHttpStorage es;
   final IndexNameFormatter indexNameFormatter;
+  private final static int MAX_CACHE_DAYS = 2;
+  private final static int MAX_CACHE_ENTRIES = 1000;
+
+  private final static Map<String, Set<Pair<String>>> indexToServiceSpansCache = Collections.synchronizedMap(new LinkedHashMap<String, Set<Pair<String>>>() {
```

Comment:
also note.. this code probably needs to be extracted to a method, because you will need to do this in the catch block, but more importantly in response to Callback.onError. You'll need to decorate the input callback with one like this...

```java
new Callback() {
  void onSuccess(){
    realCallback.onSuccess();
  }
  void onError(Throwable t){
    clearCache(indexToServiceSpans);
  }
}

```